src/components/Content/Content.tsx

```diff
@@ -144,12 +144,17 @@ const Content = (props: ContentProps): ReactElement => {
 		<MainWrapper className={cardVariant ? "card-wrapper" : "content-wrapper"}>
 			<TitleWrapper className={cardVariant ? "title-bar" : ""}>
 				<Title>{title}</Title>
-				{buttons &&
-					<ButtonsWrapper className={cardVariant ? "card-buttons" : "standard-buttons"}>
-						{buttons?.map((button, idx) => (
-							<Button key={`${button.label}-${idx}`} {...button} />
-						))}
-					</ButtonsWrapper>}
+				{buttons && (
+					<ButtonsWrapper
+						className={cardVariant ? "card-buttons" : "standard-buttons"}
+					>
+						{buttons?.map((button, idx) =>
+							button.show !== undefined ? (
+								showContent(button.show) && <Button key={`${button.label}-${idx}`} {...button} />
```

Comment:
I think we might want to memoize this outside of the render fn and just render the valid buttons, that way we don't have to run all show functions with every change.
Something like:
`const validButtons = useMemo(() => {

}, [buttons]);`

Reply:
As we were talking using arrays or objects as dependencies is not possible because of the way they are compared, there has been a proposition to be able to use a custom comparator function that would be used to determine whether the dependencies of the hook have changed, instead of the default shallow equality check. https://github.com/facebook/react/issues/14476

I made a record with the approach proposed and you can see that the useMemo computation is being made with every re-render.
![memoizing-references](https://user-images.githubusercontent.com/87880265/233711994-7b52a15f-a578-432c-8097-bb9626674d60.gif)

So I would suggest not trying to memoize this computation since the `useMemo` already has a small overhead when it checks if its dependencies changed. Moreover, I'd argue that this computation is not very expensive, and it's very unlikely that tons of buttons want to be rendered.